File: backend/slave/src/main/java/com/ociet/loadgenerator/slave/LoadResultProducer.java
package com.ociet.loadgenerator.slave;

import com.ociet.loadgenerator.common.Constants;
import com.ociet.loadgenerator.common.LoadResultMessage;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

@Service
public class LoadResultProducer {
    private static final Logger logger = LoggerFactory.getLogger(LoadResultProducer.class);

    @Autowired
    private KafkaTemplate<String, LoadResultMessage> kafkaTemplate;

    public void send(LoadResultMessage message) {
        logger.info(String.format("Producing message -> %s", message));
        this.kafkaTemplate.send(Constants.LOAD_RESULT_TOPIC, message);
    }
}

File: backend/slave/src/main/kotlin/com.chylb.loadgenerator.slave/LoadResultProducer.kt
package com.chylb.loadgenerator.slave

import com.chylb.loadgenerator.common.Constants
import com.chylb.loadgenerator.common.LoadResultMessage
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service

@Service
class LoadResultProducer {
    companion object {
        private val logger = LoggerFactory.getLogger(LoadResultProducer::class.java)
    }

    @Autowired
    private val kafkaTemplate: KafkaTemplate<String, LoadResultMessage?>? = null

    fun send(message: LoadResultMessage?) {
        logger.info(String.format("Producing message -> %s", message))
        kafkaTemplate!!.send(Constants.LOAD_RESULT_TOPIC, message)
    }
}